egs/wsj/s5/steps/cleanup/segment_long_utterances.sh

```diff
@@ -0,0 +1,398 @@
+    $dir/ctm_edits $dir/ctm_edits.modified
+
+  echo "   ... See $dir/log/modify_ctm_edits.log for details and stats, including"
+  echo " a list of commonly-repeated words."
+fi
+
+if [ $stage -le 12 ]; then
+  echo "$0: applying 'taint' markers to ctm-edits file to mark silences and"
+  echo "  ... non-scored words that are next to errors."
+  $cmd $dir/log/taint_ctm_edits.log \
+       steps/cleanup/internal/taint_ctm_edits.py --remove-deletions=false \
+       $dir/ctm_edits.modified $dir/ctm_edits.tainted
+  echo "... Stats, including global cor/ins/del/sub stats, are in $dir/log/taint_ctm_edits.log."
+fi
+
+if [ $stage -le 13 ]; then
+  echo "$0: creating segmentation from ctm-edits file."
+
+  segmentation_opts=(
+  --min-segment-length=$min_segment_length
+  --min-new-segment-length=$min_new_segment_length
+  --max-tainted-length=$max_tainted_length
+  --max-edge-silence-length=$max_edge_silence_length
+  --max-edge-non-scored-length=$max_edge_non_scored_length
+  --max-internal-silence-length=$max_internal_silence_length
+  --max-internal-non-scored-length=$max_internal_non_scored_length
+  --unk-padding=$unk_padding
+  --max-junk-proportion=$max_junk_proportion
+  --min-split-point-duration=$min_split_point_duration
+  --max-deleted-words-kept-when-merging=$max_deleted_words_kept_when_merging
+  --merging-score.silence-factor=$silence_factor
+  --merging-score.incorrect-words-factor=$incorrect_words_factor
+  --merging-score.tainted-words-factor=$tainted_words_factor
+  --merging.max-wer=$max_wer
+  --merging.max-segment-length=$max_segment_length_for_merging
+  --merging.max-bad-proportion=$max_bad_proportion
+  --merging.max-intersegment-incorrect-words-length=$max_intersegment_incorrect_words_length
+  --splitting.max-segment-length=$max_segment_length_for_splitting
+  --splitting.hard-max-segment-length=$hard_max_segment_length
+  --splitting.min-silence-length=$min_silence_length_to_split_at
+  --splitting.min-non-scored-length=$min_non_scored_length_to_split_at
+  )
+  
+  $cmd $dir/log/segment_ctm_edits.log \
+    steps/cleanup/internal/segment_ctm_edits_mild.py \
+      ${segmentation_opts[@]} \
+      --oov-symbol-file=$lang/oov.txt \
+      --ctm-edits-out=$dir/ctm_edits.segmented \
+      --word-stats-out=$dir/word_stats.txt \
+      $dir/non_scored_words.txt \
+      $dir/ctm_edits.tainted $dir/text $dir/segments
+
+  echo "$0: contents of $dir/log/segment_ctm_edits.log are:"
+  cat $dir/log/segment_ctm_edits.log
+  echo "For word-level statistics on p(not-being-in-a-segment), with 'worst' words at the top,"
+  echo "see $dir/word_stats.txt"
+  echo "For detailed utterance-level debugging information, see $dir/ctm_edits.segmented"
+fi
+
+mkdir -p $out_data
+if [ $stage -le 14 ]; then
+  utils/data/subsegment_data_dir.sh $data_uniform_seg $dir/segments $dir/text $out_data
+  
+  utils/data/get_reco2num_frames.sh --frame-shift 0.01 --frame-overlap 0.0151 \
```

Comment:
Vimal, you didn't respond to this comment.  When I ask questions like this, it does require a response.